### PR TITLE
Add element count for annotation/group combinations

### PR DESCRIPTION
**Pull request by eagw**
Adds a "Count" label in the draw widget under the draw element buttons; it gives the count of how many elements from the selected annotation are also in the currently selected group and updates it as elements are added, deleted and edited.

Addresses #174

### CODE REVIEW

**Comment by eagw** [comment]
As of now, the count updates when singular elements are altered, but I'm currently working on adding functionality so it updates when multiple elements are edited at once as well.

**Comment by eagw** [comment]
> As of now, the count updates when singular elements are altered, but I'm currently working on adding functionality so it updates when multiple elements are edited at once as well.

This should now be good to go.

**Comment by manthey** [comment]
@eagw I just tried the branch (I haven't looked at the code yet), and I see it showing one group's count.  In some instances it changes which group it shows when I cycle through the style groups by using the mouse to pick a group, but not when I use the keyboard short cuts (q and w).

I think we want to always list the counts for any group where the count is non-zero (e.g., have something like
```
Count: 10 Glomeruli
        5 Nuclei
```

**Comment by eagw** [comment]
> @eagw I just tried the branch (I haven't looked at the code yet), and I see it showing one group's count. In some instances it changes which group it shows when I cycle through the style groups by using the mouse to pick a group, but not when I use the keyboard short cuts (q and w).
> 
> I think we want to always list the counts for any group where the count is non-zero (e.g., have something like
> 
> ```
> Count: 10 Glomeruli
>         5 Nuclei
> ```

I originally implemented it so only the count of the currently selected group was shown (and forgot to make sure it was updating when shortcuts were used), but now the counts of all groups with at least one element should be visible at all times.

**Comment by manthey** [comment]
I like how this is working.

For now, we probably don't want to display counts for pixelmaps and overlay annotations.  If we do display them for pixelmaps, it really needs to be how many of each category is selected in the pixel map, not a single value for the whole pixelmap.  We can make an issue to do that in a future PR.

In the interest in using less screen real estate, would it be possible to adjust the css like so:
image
(I hacked that by putting count and the tallies in spans that were inline-block positioned, but the width is wrong).

File: histomicsui/web_client/panels/DrawWidget.js [hunk 1]
```diff
@@ -114,11 +114,16 @@ var DrawWidget = Panel.extend({
                 name,
                 opts: this._editOptions,
                 drawingType: this._drawingType,
-                collapsed: this.$('.s-panel-content.collapse').length && !this.$('.s-panel-content').hasClass('in')
+                collapsed: this.$('.s-panel-content.collapse').length && !this.$('.s-panel-content').hasClass('in'),
+                firstRender: true
             }));
             this.$('.h-dropdown-content').collapse({toggle: false});
         }
-        this.$('button.h-draw[data-type]').removeClass('active');
```
Comment (manthey) [inline histomicsui/web_client/panels/DrawWidget.js:121]: Is there a reason you removed this line?

**Comment by manthey** [comment]
You can convert this from a draft PR to a regular PR.

File: histomicsui/web_client/panels/DrawWidget.js (hunk 1 above)
Comment (eagw) [reply histomicsui/web_client/panels/DrawWidget.js:121]: No! Thank you for pointing that out!

See 0e990b1

**Review by manthey — APPROVED** [review]
Thank you.

**Pull request merged**